hyperlane/mailbox/src/execute.rs

```diff
@@ -46,6 +46,12 @@ fn dispatch(
     let cfg = CONFIG.load(ctx.storage)?;
     let (nonce, _) = NONCE.increment(ctx.storage)?;
 
+    // Ensure the destination domain is not the local domain.
+    ensure!(
+        destination_domain != cfg.local_domain,
+        "Destination domain is the same as the local domain"
```

Comment:
do not uppercase the error message. see rust API guidelines:

https://github.com/rust-lang/api-guidelines/blob/master/src/interoperability.md#examples-of-error-messages